### ISSUE TITLE
Changes to imported objects cannot be persisted

### ISSUE THREAD

**Issue by GuentherGredy** [osmandapp/OsmAnd#19892]
### Description

For my use cases I plan routes, tracks and POIs in Garmin's BaseCamp. Then I copy them to a folder on my mobile (Android 14) and import them from there to OsmAnd (4.7.10); folder "...\Interner Speicher\Android\data\net.osmand.plus\files\tracks\ActualTour".
In former versions of OsmAnd I could change properties of POIs (color, icon) and persist the changes. In the newest version this is not possible any more.

### Steps to reproduce

Menu: Meine Orte
Tab: Tracks
Selection: folder "ActualTour"
Menu: Import
Selection: POI on map
Press "Aktionen"
Press "GPX-Wegpunkt bearbeiten"
Select icon and color
Press "Speichern"

### Actual result

POI remains unchanged

### Expected result

POI gets the edited properties

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: 4.7.10
Android/iOS version: 14
Device model: Samsung Galaxy A54
Crash-Logs: ?
```

**Comment by yuriiurshuliak** [comment]
Could you provide the tracks and POIs where you can't change the POI properties, so we can test it on our end?

**Comment by yuriiurshuliak** [comment]
The bug could not be reproduced. To reproduce the issue, import your waypoints into the tracks folder. Then, view them on the map, click on the icon, and try to edit the GPX. Change the icon and color, and observe if the changes are successfully saved and reflected on the map.

OsmAnd~ 4.8.0#2416m, released: 2024-05-30

https://github.com/osmandapp/OsmAnd/assets/127092082/3ecda9cd-bcf0-43fd-8eb7-020df668c9c7

**Comment by sonora** [comment]
@yuriiurshuliak  Now restart the app and check if the appearance change is not lost.

I actually observe a very similar issue with a slightly different procedure: For a track containing only waypoints, display it and on its context menu, tap on the 'Points' button, then on the 3-dots menu for any waypoint category, use "Change default appeatance", and set to to something different. That change works only until the next app start. (It seems that after that the appearance of the waypoint icons changes to the _default appearance for favorites??_)

**Comment by sonora** [comment]
PS: Inspecting a gpx file where in OsmAnd I tried to change the waypoint icon color to _blue_ resulted in these lines (with incorrect color value) being added to all wpt's:
```
    <extensions>
      <osmand:color>#00000000</osmand:color>
    </extensions>
```

**Comment by vshcherb** [comment]
To do:
- Changing default appearance or appearance shouldn't add any colors or tags inside GPX.
- Tags couldn't only change if you "Edit track" in Plan Route.
- All the changes of appearance should be inside database

**Comment by sonora** [comment]
@vshcherb Upon re-reading the todo's, I am a little unclear if/when we actually transfer appearance changes to the gpx files, vs. only note that in the gpx database? What's our strategy here, does e.g. "Reset should delete `osmand:points_groups`" mean we modify the file itself?

**Comment by vshcherb** [comment]
Yes, we don't store point appearance in DB cause it's too much to handle for now, so we will modify file but only a specific section for waypoint appearance

**Comment by fjb2020** [comment]
If it's helpful, I'm seeing what I think is the same issue - changes to gpx track appearance (colour, width etc) not persisting across app restarts. Specifically it is happening on the OsmAnd Nightly 4.9.0#3081m release on my Galaxy A53 Android 14. The bug does not seem to be present on the production OsmAnd+ 4.8.6 on the same device using the same gpx track

**Comment by sonora** [comment]
To put some more clarity in here: I have meanwhile found out that some of my numbered issues above (3, 4, 5) are fixed by properly announcing the `osmand:` extensions, i.e. by adding `mlns:osmand="https://osmand.net"` in the opeing `gpx` tag (line 2 of most files).

So that may reduce the items to check/do in this issue to:
- [ ] Check if we have code to add `mlns:osmand="https://osmand.net"` in all cases where a user modifies wpt or Category appearance and it is still missing for some reason
- [ ] Investigate why in some cases #00000000 is used as a falback color, there is some issue with using the correct points_groups color value (to my research only in the tracks code, it works in the favorites code. And in the tracks code it is correctly detected, just not applied.)
- [ ] Make sure that upon users' appearance modifications we add individual wpt appearance to the gpx file only for wpts and tags which differ from their points_groups value, not per se to all wpts.

**Comment by vshcherb** [comment]
Getting back to this issue, need to retest and clarify. 
> (1) Changes to track line appearance seem indeed not written back to the gpx file, hence seem lost after app restart. Probably a regression

Track line changes do not write to track file, but they appear after restart.
